packages/react-import-remote/src/test/ImportRemote.test.tsx

```diff
@@ -123,37 +147,47 @@ describe('<ImportRemote />', () => {
   });
 
   describe('preconnect', () => {
-    it('does not render any preconnect link by default', () => {
+    it('does not render any preconnect link by default', async () => {
+      const resolve = createResolvableLoad();
       const importRemote = mount(<ImportRemote {...mockProps} />);
 
+      await importRemote.act(async () => {
+        await resolve();
+      });
+
       expect(importRemote).not.toContainReactComponent(Preconnect);
     });
 
-    it('creates a preconnect link with the source’s origin when preconnecting is requested', () => {
+    it('creates a preconnect link with the source’s origin when preconnecting is requested', async () => {
+      const resolve = createResolvableLoad();
       const importRemote = mount(<ImportRemote {...mockProps} preconnect />);
 
+      await importRemote.act(async () => {
+        await resolve();
+      });
+
       expect(importRemote).toContainReactComponent(Preconnect, {
         source: new URL(mockProps.source).origin,
       });
     });
   });
 
   describe('defer', () => {
-    it('does not call load until idle when defer is DeferTiming.Idle', () => {
+    it('does not call load until idle when defer is DeferTiming.Idle', async () => {
       const importRemote = mount(
         <ImportRemote {...mockProps} defer={DeferTiming.Idle} />,
       );
 
       expect(load).not.toHaveBeenCalled();
 
-      importRemote.act(() => {
-        requestIdleCallback.runIdleCallbacks();
+      await importRemote.act(async () => {
+        await requestIdleCallback.runIdleCallbacks();
```

Comment:
Does this return a promise?

Reply:
No, but without the above async/awaits we get the act console warnings.

Reply:
We should be able to await on the result of calling `load` instead then?